src/differentiable_functions.jl

```diff
@@ -113,3 +113,22 @@ function DifferentiableSparseMultivariateFunction(f!::Function, g!::Function)
     end
     return DifferentiableSparseMultivariateFunction(f!, g!, fg!)
 end
+
+
+immutable DifferentiableGivenSparseMultivariateFunction{Tv, Ti} <: AbstractDifferentiableMultivariateFunction
+    f!::Function
+    g!::Function
+    fg!::Function
+    J::SparseMatrixCSC{Tv, Ti}
+end
+
+alloc_jacobian(df::DifferentiableGivenSparseMultivariateFunction, args...) = deepcopy(df.J)
```

Comment:
Shouldn't a normal copy work? 

Whether it is safe or not to modify the input matrix or not is debatable. Personally I wouldn't mind it being modified because the reason I created the sparse matrix in the first place is to use it in the solver.